tests/check_mode/mod.rs

```diff
@@ -0,0 +1,208 @@
+// Common test utilities shared across test modules
+
+use std::path::{Path, PathBuf};
+use std::process::{Command, Output};
+
+/// Gets the path to a fixture directory
+pub fn fixture_path(name: &str) -> PathBuf {
+    Path::new("tests").join("fixtures").join(name)
+}
+
+/// Run cargo-autoinherit in the specified directory with given args
+pub fn run_autoinherit<P: AsRef<Path>>(dir: P, args: &[&str]) -> Output {
+    Command::new("cargo")
+        .arg("autoinherit")
+        .args(args)
+        .current_dir(dir)
+        .output()
+        .expect("Failed to execute cargo-autoinherit")
+}
```

Comment:
This is going to run the currently installed version of cargo-autoinherit, it would probably make more sense to build and run straight from the code.

I'm not sure if this is the best possible way, but I usually use https://crates.io/crates/escargot for doing that kind of stuff.

Reply:
I don't actually have the tests passing yet, but it should be something like this: 

```suggestion
/// Run cargo-autoinherit in the specified directory with given args
pub fn run_autoinherit<P: AsRef<Path>>(dir: P, args: &[&str]) -> Output {
    escargot::CargoBuild::new()
        .package("cargo-autoinherit")
        .run()
        .unwrap()
        .command()
        .arg("autoinherit")
        .args(args)
        .current_dir(dir)
        .stdout(Stdio::inherit())
        .stderr(Stdio::inherit())
        .output()
        .expect("Failed to execute cargo-autoinherit")
}
```

Reply:
ah great catch, thank you!

Reply:
I suspect in `assert_cmd` it'd be:

```rust
pub fn run_autoinherit<P: AsRef<Path>>(dir: P, args: &[&str]) -> Output {
    let mut cmd = assert_cmd::Command::cargo_bin("cargo-autoinherit").unwrap();

    cmd.arg("autoinherit")
        .args(args)
        .current_dir(dir);

    cmd.output()
        .success()
}
```

Reply:
That really got rid of a lot of stdout/stderr processing code :sparkles:  :broom: